### PR TITLE
Fix redundant filter in sparql queries

### LINKED ISSUE #562: When specifying namespaces, redundant FILTER condition is introduced in SPARQL queries
For example:

```
  FILTER(isIRI(?term) && (STRSTARTS(str(?term), "http://purl.obolibrary.org/obo/UPHENO_") || STRSTARTS(str(?term), "http://purl.obolibrary.org/obo/UPHENO_")))
```

### PR DESCRIPTION
**Pull request by anitacaron**
Fixes #562

### CODE REVIEW

File: template/_dynamic_files.jinja2 [hunk 1]
```diff
@@ -399,7 +399,7 @@ WHERE {
   { ?s1 ?p1 ?term . }
   UNION
   { ?term ?p2 ?o2 . }
-  FILTER(isIRI(?term) && (STRSTARTS(str(?term), "http://purl.obolibrary.org/obo/{{ project.id.upper() }}_"){% if project.namespaces %}{% for ns in project.namespaces %} || STRSTARTS(str(?term), "{{ ns }}"){% endfor %}{% endif %}))
+  FILTER(isIRI(?term) && ({% if project.namespaces %}{% for ns in project.namespaces %}STRSTARTS(str(?term), "{{ ns }}"){% if not loop.last %} || {% endif %}{% endfor %}{% else %}STRSTARTS(str(?term), "http://purl.obolibrary.org/obo/{{ project.id.upper() }}_"){% endif %}))
```
Comment (matentzn) [inline template/_dynamic_files.jinja2:402]: How awesome is `{% if not loop.last %}` ?????

Sweeeet!

**Review by matentzn — APPROVED** [review]
Excellent, thank you @anitacaron

I assume you tested this :D

File: template/_dynamic_files.jinja2 (hunk 1 above)
Comment (anitacaron) [reply template/_dynamic_files.jinja2:402]: I just found out about it too 😄

**Comment by anitacaron** [comment]
Yes, I tested with and without namespaces :)

**Pull request merged**